common/graph/choose.go

```diff
@@ -61,5 +61,5 @@ func choose(n int, targetAmount int, i int, currentSubGroup []int, subGroups *or
 	// We either pick the current element
 	choose(n, targetAmount, i+1, append(currentSubGroup, i), subGroups)
 	// Or don't pick it
-	choose(n, targetAmount, i+1, currentSubGroup, subGroups)
```

Comment:
Nice catch and thanks for the fix.

I think it reads better if we clone when we pick the element and not when we don't pick it:

```
	// We either pick the current element
	choose(n, targetAmount, i+1, concatInts(currentSubGroup, i), subGroups)
	// Or don't pick it
	choose(n, targetAmount, i+1, currentSubGroup, subGroups)
```

and let's also move the make([]int, 0) to a separate function so it will read better, like:

```
func concatInts(a []int, elements  ... int) []int {
	var res []int
	res = append(res, a ...)
	for _, e := range elements {
		res = append(res, e)
	}

	return res
}
```